app/src/main/java/com/github/sdpteam15/polyevents/view/adapter/EventEditAdminAdapter.kt

```diff
@@ -0,0 +1,118 @@
+package com.github.sdpteam15.polyevents.view.adapter
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.view.View
+import android.view.View.INVISIBLE
+import android.view.View.VISIBLE
+import android.view.ViewGroup
+import android.widget.ImageButton
+import android.widget.TextView
+import androidx.lifecycle.LifecycleOwner
+import androidx.recyclerview.widget.RecyclerView
+import com.github.sdpteam15.polyevents.R
+import com.github.sdpteam15.polyevents.model.entity.Event
+import com.github.sdpteam15.polyevents.model.entity.MaterialRequest
+import com.github.sdpteam15.polyevents.model.observable.ObservableList
+import com.github.sdpteam15.polyevents.model.observable.ObservableMap
+import java.time.format.DateTimeFormatter
+
+/**
+ * Recycler Adapter for the list of event edit request
+ * @param context context of parent view used to inflate new views
+ * @param lifecycleOwner parent to enable observables to stop observing when the lifecycle is closed
+ * @param events List of all events edit request
+ * @param origEvent A map that map the eventId to the original event (the one already in database)
+ * @param onAcceptListener Listener for the click on the "accept" button
+ * @param onRefuseListener Listener for the click on the "refuse" button
+ * @param onSeeListener Listener for the click on the "see" button
+ */
+class EventEditAdminAdapter(
+    context: Context,
+    lifecycleOwner: LifecycleOwner,
+    private val events: ObservableList<Event>,
+    private val origEvent: ObservableMap<String, Event>,
+    private val onAcceptListener: (Event) -> Unit,
+    private val onRefuseListener: (Event) -> Unit,
+    private val onSeeListener: (Event, Boolean, Event?) -> Unit,
+) : RecyclerView.Adapter<EventEditAdminAdapter.ItemViewHolder>() {
+    private val adapterLayout = LayoutInflater.from(context)
+
+    // Add observer to both list to update the recyclerview content when the lists are updated
+    init {
+        events.observe(lifecycleOwner) {
+            notifyDataSetChanged()
+        }
+
+        origEvent.observe(lifecycleOwner) {
+            notifyDataSetChanged()
+        }
+    }
```

Comment:
Just to be sure, these are not listeners on changes in the remote database no? Which changes do they react to?

Reply:
Nop! They react on changes made on the observable lists (by teh database) and click on button (for accept, delete, etc...)